TopMovies/Resources/Assembler/Assembler.swift

```diff
@@ -0,0 +1,46 @@
+//
+//  Assembler.swift
+//  TopMovies
+//
+//  Created by Macbook Pro  on 19.03.2021.
+//
+
+import Swinject
+import ReSwift
+import Overture
+
+typealias MainStore = Store<MainState>
+
+struct Assembler {
+  static func registerDependencies() {
+    let registrar = Container.default.registrar
+    
+    // MARK: - Store
+    registrar.register(MainStore.self) { _ in
+      MainStore(reducer: mainReducer, state: nil, middleware: [allActionsMiddleware])
+    }.inObjectScope(.container)
+    
+    // MARK: - Service
+    registrar.register(MovieAPIProtocol.self) { _ in MovieAPI() }
+    registrar.register(MovieService<StoreProvider<MainState>>.self) { resolver in
+      MovieService(
+        movieAPI: resolver.resolve(MovieAPIProtocol.self)!,
+        storeProvider: curry(StoreProvider<MainState>.init(store: onStateUpdate:))(resolver.resolve(MainStore.self)!)
+      )
+    }.inObjectScope(.container)
+    
+    // MARK: - Factory
+    registrar.register(VCFactoryProtocol.self) { _ in
+      VCFactory()
+    }
+    registrar.register(ConnectorFactoryProtocol.self) { _ in
+      ConnectorFactory()
```

Comment:
You can pass init:
`registrar.register(ConnectorFactoryProtocol.self, ConnectorFactory.init)`
:)

Reply:
> You can pass init:
> `registrar.register(ConnectorFactoryProtocol.self, ConnectorFactory.init)`
> :)

ConnectorFactory.init is () -> ConnectorFactory but method register(_: , factory:) expects (resolver) -> ConnectorFactory type in 'factory' parameter

Reply:
Hmmm... You could use AutoRegister for that:
https://github.com/Swinject/SwinjectAutoregistration